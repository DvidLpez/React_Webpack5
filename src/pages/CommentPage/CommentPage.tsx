import React, { FC, Fragment, ReactElement, useEffect } from "react";
import { RouteComponentProps, useLocation } from 'react-router-dom';
import { useDispatch, useSelector } from "react-redux";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faUser, faComments, faCalendarAlt } from "@fortawesome/free-regular-svg-icons";
import { faExternalLinkAlt } from "@fortawesome/free-solid-svg-icons";
import moment from "moment";
import { loadIssueAction } from "../../redux/actions/issuesAction";
import ButtomHome from "../../components/ButtomHome/ButtomHome";
import CommentsList from "../../components/CommentsList/CommentsList";
import Header from "../../components/Header/Header";
import Loading from "../../components/Loading/Loading";
import Error from "../../components/Error/Error";
import IAppState  from "../../interfaces/IAppState";
import IIssue from '../../interfaces/IIssue';
import { Wrapper, Span, Text, Subtitle, LineHr } from "../../components/StyledComponents/styledComponents";
import { Info, WrapperInfo, WrapperProblem, TextState} from "./Styles";

type TCommentsParams = {
   number: string;
}

const CommentPage: FC<RouteComponentProps<TCommentsParams>> = ({ match }: RouteComponentProps<TCommentsParams>): ReactElement => {
   
   const { pathname } = useLocation();

   useEffect(() => {
      window.scrollTo(0, 0);
   }, [pathname]);

   const number = match.params.number;
   const { data, error, loading } = useSelector((state: IAppState) => state.issues);
   const issue: IIssue = data.filter((issue: IIssue) => issue.number == parseInt(number) )[0];
   
   const dispatch = useDispatch();

   useEffect(() => {

      if (!issue) {
         dispatch(loadIssueAction(parseInt(number)));
      }

   }, [dispatch, issue, number]);

   return (
      <Fragment>
         <Header title="Issue comments"/>
         <ButtomHome />
         { error ? <Error />: null }
         { loading ? <Loading /> : null }
         { issue ? 
               <Wrapper>
                  <WrapperProblem>
                     <WrapperInfo>
                        <Info>
                           <TextState state={issue.state.toLocaleLowerCase()}>{issue.state}</TextState>
                        </Info>
                        <Info>
                           <FontAwesomeIcon icon={faUser} /> <Text>{issue.author?.login}</Text>
                        </Info>
                        <Info>
                           <FontAwesomeIcon icon={faCalendarAlt} /><Text> {moment(issue.createdAt).format('DD/MM/YYYY HH:mm')}</Text>
                        </Info>
                        <Info>
                           <FontAwesomeIcon icon={faComments} /><Text> {issue.comments.totalCount} </Text>
                        </Info>
                        <Info>
                           <a href={issue.url} target="_blank" rel="noreferrer">
                           <FontAwesomeIcon icon={faExternalLinkAlt} /> <Text>See in Github</Text> 
                           </a>
                        </Info>
                     </WrapperInfo>
                     <LineHr />
                     <Subtitle>{issue.title} <Span>#{issue.number}</Span></Subtitle>
                     <div className="comment_html" dangerouslySetInnerHTML={{ __html: issue.bodyHTML }} />
                  </WrapperProblem>
                  <CommentsList comments={issue.comments}/>
               </Wrapper>
         : null}
      </Fragment>
   )
}
export default CommentPage;
