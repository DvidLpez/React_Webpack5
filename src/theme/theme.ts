import ITheme from "../interfaces/ITheme";

export const theme: ITheme = {
  fonts: {
    style: {
      primary: "'Source Sans Pro', sans-serif",
    },
    size: {
      small: "0.9em",
      default: "1em",
      medium: "1.2em",
      large: "1.6em",
      big: "2em",
      extra: "3em",
      extraBig: "5em",
    },
    weight: {
      bold: "600",
    },
  },
  colors: {
    white: "#fff",
    lightgrey: "#7c7c7c",
    link: "#000",
    border: "#dbdbdb",
    labelSearch: "#377d6a",
    placeholder: "#cccccc",
    success: "#008000",
    closed: "#c60404",
    danger: "#932323",
    bgBody: "#fafafa",
    bgButtons: "#414141",
    bgComment: "#fff9d6",
    gradient: {
      bgHeader:
        "linear-gradient(to right, #3e9e7e 0%, #2e755d 10%, #205040 20%, #414141 30%)",
    },
    shadow: {
      linkHome: "2px 4px 11px 1px rgba(138, 138, 138, 1)",
      inputSearch: "0 0 20px 5px rgba(163, 163, 163, 1)",
      labelSearch: "0 1px 0 rgba(19, 74, 70, 0.4)",
    },
  },
  spaces: {
    none: "0",
    center: "center",
    button: "10px 40px",
    borderRadius: "5px",
    s5: "5px",
    s10: "10px",
    s15: "15px",
    s20: "20px",
    s25: "25px",
    s30: "30px",
    s50: "50px",
    p55: "5px 5px",
    p100: "10px 0",
    p1520: "15px 20px",
    p1010: "10px 10px",
    p1515: "15px 15px",
    p2020: "20px 20px",
    p2525: "25px 25px",
    p3030: "30px 30px",
  },
};
