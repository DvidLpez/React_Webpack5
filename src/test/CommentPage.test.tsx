import React from "react";
import { mount, shallow, render } from 'enzyme';
import { ThemeProvider } from "styled-components";
import { theme } from "../theme/theme";
import { MemoryRouter } from 'react-router-dom';
import CommentPage from "../pages/CommentPage/CommentPage";
import { Provider } from "react-redux";
import store from "../redux/store";
import * as reactRedux from 'react-redux';
import IAppState from "../interfaces/IAppState";

describe('TESTING COMMENT PAGE', () => {

   const useSelectorMock = jest.spyOn(reactRedux, 'useSelector')

   it('*** Check Comment Page ***', () => {

      beforeEach(() => {
         useSelectorMock.mockClear()
      })

      useSelectorMock.mockReturnValue({ data: [
         { 
            number: '21098', 
            state: 'closed',
            comments: {
               totalCount: 3,
               nodes: [{
                  bodyHTML: "<h1>Test Comment Page with Enzyme</h1>",
                  author: {
                     login: 'David LÃ³pez',
                  },
                  createdAt: "2021-03-25T15:03:36Z"
               }]
            }
         }
      ]})

      const routeComponentPropsMock = {
         // add jest.fn() as needed to any of the objects
         history: {} as any,
         location: {} as any,
         match: { params: { number: '21098'}} as any
      }
     
      const wrapper = render(
         <Provider store={store}>
             <ThemeProvider theme={theme}>
               <MemoryRouter>
                  <CommentPage  {...routeComponentPropsMock}/>
               </MemoryRouter>
            </ThemeProvider>
         </Provider>
      )
      expect(wrapper).toMatchSnapshot()
   })

})