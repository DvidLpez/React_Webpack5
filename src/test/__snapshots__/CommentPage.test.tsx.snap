// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING COMMENT PAGE - Render page if has an error 1`] = `
Array [
  <div
    class="sc-bdfBwQ sc-iqHYGH kUdDos jnqbg"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-github fa-w-16 icon_brand"
      data-icon="github"
      data-prefix="fab"
      focusable="false"
      role="img"
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
        fill="currentColor"
      />
    </svg>
    <h1
      class="sc-gsTCUz sc-kEjbxe jrFgNj elJGYe"
    >
      Issue comments
    </h1>
  </div>,
  <div
    class="sc-bdfBwQ sc-fubCfw kUdDos fwLbUW"
  >
    <a
      class="sc-gKsewC fMfLPa"
      href="/"
    >
      Go Home
    </a>
  </div>,
  <div
    class="sc-bdfBwQ sc-kstrdz kUdDos hPtqPh"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-bomb fa-w-16 icon_error"
      data-icon="bomb"
      data-prefix="fas"
      focusable="false"
      role="img"
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M440.5 88.5l-52 52L415 167c9.4 9.4 9.4 24.6 0 33.9l-17.4 17.4c11.8 26.1 18.4 55.1 18.4 85.6 0 114.9-93.1 208-208 208S0 418.9 0 304 93.1 96 208 96c30.5 0 59.5 6.6 85.6 18.4L311 97c9.4-9.4 24.6-9.4 33.9 0l26.5 26.5 52-52 17.1 17zM500 60h-24c-6.6 0-12 5.4-12 12s5.4 12 12 12h24c6.6 0 12-5.4 12-12s-5.4-12-12-12zM440 0c-6.6 0-12 5.4-12 12v24c0 6.6 5.4 12 12 12s12-5.4 12-12V12c0-6.6-5.4-12-12-12zm33.9 55l17-17c4.7-4.7 4.7-12.3 0-17-4.7-4.7-12.3-4.7-17 0l-17 17c-4.7 4.7-4.7 12.3 0 17 4.8 4.7 12.4 4.7 17 0zm-67.8 0c4.7 4.7 12.3 4.7 17 0 4.7-4.7 4.7-12.3 0-17l-17-17c-4.7-4.7-12.3-4.7-17 0-4.7 4.7-4.7 12.3 0 17l17 17zm67.8 34c-4.7-4.7-12.3-4.7-17 0-4.7 4.7-4.7 12.3 0 17l17 17c4.7 4.7 12.3 4.7 17 0 4.7-4.7 4.7-12.3 0-17l-17-17zM112 272c0-35.3 28.7-64 64-64 8.8 0 16-7.2 16-16s-7.2-16-16-16c-52.9 0-96 43.1-96 96 0 8.8 7.2 16 16 16s16-7.2 16-16z"
        fill="currentColor"
      />
    </svg>
    <p
      class="sc-eCssSg gGMhBG"
    >
      Opps!! We have an error. Try again!!
    </p>
  </div>,
]
`;

exports[`TESTING COMMENT PAGE - Render page if is loading 1`] = `
Array [
  <div
    class="sc-bdfBwQ sc-iqHYGH kUdDos jnqbg"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-github fa-w-16 icon_brand"
      data-icon="github"
      data-prefix="fab"
      focusable="false"
      role="img"
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
        fill="currentColor"
      />
    </svg>
    <h1
      class="sc-gsTCUz sc-kEjbxe jrFgNj elJGYe"
    >
      Issue comments
    </h1>
  </div>,
  <div
    class="sc-bdfBwQ sc-fubCfw kUdDos fwLbUW"
  >
    <a
      class="sc-gKsewC fMfLPa"
      href="/"
    >
      Go Home
    </a>
  </div>,
  <div
    class="sc-bdfBwQ sc-crrsfI kUdDos hyZdiz"
  >
    <div
      class="sc-dQppl dzisxz"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-spinner fa-w-16 "
        data-icon="spinner"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M304 48c0 26.51-21.49 48-48 48s-48-21.49-48-48 21.49-48 48-48 48 21.49 48 48zm-48 368c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm208-208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zM96 256c0-26.51-21.49-48-48-48S0 229.49 0 256s21.49 48 48 48 48-21.49 48-48zm12.922 99.078c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.491-48-48-48zm294.156 0c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48c0-26.509-21.49-48-48-48zM108.922 60.922c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.491-48-48-48z"
          fill="currentColor"
        />
      </svg>
    </div>
    <p
      class="sc-eCssSg sc-bqyKva gGMhBG hpDYSa"
    >
      Loading
    </p>
  </div>,
]
`;

exports[`TESTING COMMENT PAGE - Render page with issue and comments 1`] = `
Array [
  <div
    class="sc-bdfBwQ sc-iqHYGH kUdDos jnqbg"
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-github fa-w-16 icon_brand"
      data-icon="github"
      data-prefix="fab"
      focusable="false"
      role="img"
      viewBox="0 0 496 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
        fill="currentColor"
      />
    </svg>
    <h1
      class="sc-gsTCUz sc-kEjbxe jrFgNj elJGYe"
    >
      Issue comments
    </h1>
  </div>,
  <div
    class="sc-bdfBwQ sc-fubCfw kUdDos fwLbUW"
  >
    <a
      class="sc-gKsewC fMfLPa"
      href="/"
    >
      Go Home
    </a>
  </div>,
  <div
    class="sc-bdfBwQ kUdDos"
  >
    <div
      class="sc-bdfBwQ sc-hBEYos kUdDos keQKtR"
    >
      <div
        class="sc-bdfBwQ sc-fodVxV kUdDos bUdnMX"
      >
        <div
          class="sc-bdfBwQ sc-fFubgz kUdDos bdiIWK"
        >
          <p
            class="sc-eCssSg sc-bkzZxe gGMhBG OHRcp"
          >
            closed
          </p>
        </div>
        <div
          class="sc-bdfBwQ sc-fFubgz kUdDos bdiIWK"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-user fa-w-14 "
            data-icon="user"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M313.6 304c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 304 0 364.2 0 438.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-25.6c0-74.2-60.2-134.4-134.4-134.4zM400 464H48v-25.6c0-47.6 38.8-86.4 86.4-86.4 14.6 0 38.3 16 89.6 16 51.7 0 74.9-16 89.6-16 47.6 0 86.4 38.8 86.4 86.4V464zM224 288c79.5 0 144-64.5 144-144S303.5 0 224 0 80 64.5 80 144s64.5 144 144 144zm0-240c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
              fill="currentColor"
            />
          </svg>
           
          <p
            class="sc-eCssSg gGMhBG"
          >
            David LÃ³pez
          </p>
        </div>
        <div
          class="sc-bdfBwQ sc-fFubgz kUdDos bdiIWK"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-calendar-alt fa-w-14 "
            data-icon="calendar-alt"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M148 288h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm108-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 96v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm192 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96-260v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
              fill="currentColor"
            />
          </svg>
          <p
            class="sc-eCssSg gGMhBG"
          >
             22/03/2021 16:03
          </p>
        </div>
        <div
          class="sc-bdfBwQ sc-fFubgz kUdDos bdiIWK"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-comments fa-w-18 "
            data-icon="comments"
            data-prefix="far"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M532 386.2c27.5-27.1 44-61.1 44-98.2 0-80-76.5-146.1-176.2-157.9C368.3 72.5 294.3 32 208 32 93.1 32 0 103.6 0 192c0 37 16.5 71 44 98.2-15.3 30.7-37.3 54.5-37.7 54.9-6.3 6.7-8.1 16.5-4.4 25 3.6 8.5 12 14 21.2 14 53.5 0 96.7-20.2 125.2-38.8 9.2 2.1 18.7 3.7 28.4 4.9C208.1 407.6 281.8 448 368 448c20.8 0 40.8-2.4 59.8-6.8C456.3 459.7 499.4 480 553 480c9.2 0 17.5-5.5 21.2-14 3.6-8.5 1.9-18.3-4.4-25-.4-.3-22.5-24.1-37.8-54.8zm-392.8-92.3L122.1 305c-14.1 9.1-28.5 16.3-43.1 21.4 2.7-4.7 5.4-9.7 8-14.8l15.5-31.1L77.7 256C64.2 242.6 48 220.7 48 192c0-60.7 73.3-112 160-112s160 51.3 160 112-73.3 112-160 112c-16.5 0-33-1.9-49-5.6l-19.8-4.5zM498.3 352l-24.7 24.4 15.5 31.1c2.6 5.1 5.3 10.1 8 14.8-14.6-5.1-29-12.3-43.1-21.4l-17.1-11.1-19.9 4.6c-16 3.7-32.5 5.6-49 5.6-54 0-102.2-20.1-131.3-49.7C338 339.5 416 272.9 416 192c0-3.4-.4-6.7-.7-10C479.7 196.5 528 238.8 528 288c0 28.7-16.2 50.6-29.7 64z"
              fill="currentColor"
            />
          </svg>
          <p
            class="sc-eCssSg gGMhBG"
          >
             1 
          </p>
        </div>
        <div
          class="sc-bdfBwQ sc-fFubgz kUdDos bdiIWK"
        >
          <a
            href="https://lopezblasco.com"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-external-link-alt fa-w-16 "
              data-icon="external-link-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M432,320H400a16,16,0,0,0-16,16V448H64V128H208a16,16,0,0,0,16-16V80a16,16,0,0,0-16-16H48A48,48,0,0,0,0,112V464a48,48,0,0,0,48,48H400a48,48,0,0,0,48-48V336A16,16,0,0,0,432,320ZM488,0h-128c-21.37,0-32.05,25.91-17,41l35.73,35.73L135,320.37a24,24,0,0,0,0,34L157.67,377a24,24,0,0,0,34,0L435.28,133.32,471,169c15,15,41,4.5,41-17V24A24,24,0,0,0,488,0Z"
                fill="currentColor"
              />
            </svg>
             
            <p
              class="sc-eCssSg gGMhBG"
            >
              See in Github
            </p>
          </a>
        </div>
      </div>
      <hr
        class="sc-iBPRYJ jUHYqe"
      />
      <h2
        class="sc-dlfnbm iVxDXV"
      >
        Jest/Enzyme comment page test 
        <span
          class="sc-jSgupP xdqzF"
        >
          #21098
        </span>
      </h2>
      <div
        class="comment_html"
      >
        <h1>
          Testing page with jest/enzyme
        </h1>
      </div>
    </div>
    <h3
      class="sc-hKgILt fTcPCv"
    >
      *** Comments ***
    </h3>
    <div
      class="sc-bdfBwQ sc-pFZIQ kUdDos htphup"
    >
      <div>
        <h2>
          Test Comment Page with Enzyme
        </h2>
      </div>
      <hr
        class="sc-iBPRYJ jUHYqe"
      />
      <p
        class="sc-eCssSg sc-jrAGrp gGMhBG GAUvp"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-user fa-w-14 "
          data-icon="user"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M313.6 304c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 304 0 364.2 0 438.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-25.6c0-74.2-60.2-134.4-134.4-134.4zM400 464H48v-25.6c0-47.6 38.8-86.4 86.4-86.4 14.6 0 38.3 16 89.6 16 51.7 0 74.9-16 89.6-16 47.6 0 86.4 38.8 86.4 86.4V464zM224 288c79.5 0 144-64.5 144-144S303.5 0 224 0 80 64.5 80 144s64.5 144 144 144zm0-240c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
            fill="currentColor"
          />
        </svg>
         David LÃ³pez
      </p>
      <p
        class="sc-eCssSg sc-jrAGrp gGMhBG kqDZAq"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-calendar-alt fa-w-14 "
          data-icon="calendar-alt"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M148 288h-40c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12zm108-12v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 96v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm-96 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm192 0v-40c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12zm96-260v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V112c0-26.5 21.5-48 48-48h48V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h128V12c0-6.6 5.4-12 12-12h40c6.6 0 12 5.4 12 12v52h48c26.5 0 48 21.5 48 48zm-48 346V160H48v298c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
            fill="currentColor"
          />
        </svg>
         25/03/2021 16:03
      </p>
    </div>
  </div>,
]
`;
